arch/arm64/include/kernel_arch_thread.h

```diff
@@ -0,0 +1,52 @@
+/*
+ * Copyright (c) 2019 Carlo Caione <ccaione@baylibre.com>
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+/**
+ * @file
+ * @brief Per-arch thread definition
+ *
+ * This file contains definitions for
+ *
+ *  struct _thread_arch
+ *  struct _callee_saved
+ *
+ * necessary to instantiate instances of struct k_thread.
+ */
+
+#ifndef ZEPHYR_ARCH_ARM64_INCLUDE_KERNEL_ARCH_THREAD_H_
+#define ZEPHYR_ARCH_ARM64_INCLUDE_KERNEL_ARCH_THREAD_H_
+
+#ifndef _ASMLANGUAGE
+#include <zephyr/types.h>
+
+struct _callee_saved {
```

Comment:
what uses this? since you do not put it in _thread_arch

Reply:
ACK (failed to `git add -p`)

Reply:
this is actually needed because it is used into `struct k_thread`.